### ISSUE TITLE
`--keep-prefix` should mark prefix as invalid

### ISSUE THREAD

**Issue by eschnett** [LLNL/spack#1149]
I just received this warning from Spack:

```
  Spack will think this package is installed. Manually remove this
  directory to fix:
  /ddnA/project/eschnett/shelob/src/spack/opt/spack/linux-
  redhat6-x86_64/gcc-6.1.0-spack/julia-master-
  65fs5pr57mv6jfcj7cqvu3ipdw4n56uo
```

Instead of asking me to manually remove the prefix, Spack should mark the prefix directory as "incomplete" (e.g. via a file named `THIS IS NOT A COMPLETE INSTALL"), and should test for this file while checking whether the prefix is a valid install.

**Comment by adamjstewart** [comment]
What exactly is `--keep-prefix` used for? I've never seen a use for it myself.

**Comment by tgamblin** [comment]
If you have a package that writes things to the prefix before it is done installing, you can use this to see what happened when you build would otherwise fail.  Its for debugging a build.  The idea is really that you take the burden of having a corrupt install directory on yourself when you use it.

@eschnett: what do you use it for?

**Comment by eschnett** [comment]
I use it for debugging -- my Julia install is giving errors, and I need to see the details to solve them.

When I run `spack install julia` again, then it would be convenient if Spack deleted the directory before installing, instead of aborting with an error. Since Spack could easily determine that the prefix was just left for debugging, it might as well do so.

**Comment by citibeth** [comment]
It's used as a part of `spack setup` (#543).  Maybe there's a better way to do this.  Anyway, I get that message fairly frequently, and then I delete the directory as it suggests.  Annoying, yes, but not the end of the world.

This happens with `spack setup` because it does a half-build, then dumps you on your own with `spconfig.py`.  You may or may not ever finish the build.  But you can if you want, with `../spconfig.py ..` and then `make install`.

**Comment by becker33** [comment]
I believe this is fixed by #4331. Feel free to reopen if that is not true.
